File: Java/02- Relacionando Classes/15- Abstract e ToString/Juridica.java

package com.mycompany.mavenproject5;


public class Juridica extends Pessoa{
    
    private String cnpj;
    private String inscricaoEstadual;

    public Juridica(String cnpj, String inscricaoEstadual, String nome, String telefone, String email, Endereco endereco) {
        super(nome, telefone, email, endereco);
        this.cnpj = cnpj;
        this.inscricaoEstadual = inscricaoEstadual;
    }

    public String getInscricaoEstadual() {
        return inscricaoEstadual;
    }

    public void setInscricaoEstadual(String inscricaoEstadual) {
        this.inscricaoEstadual = inscricaoEstadual;
    }

    public String getCnpj() {
        return cnpj;
    }

    public void setCnpj(String cnpj) {
        this.cnpj = cnpj;
    }

    @Override
    public String toString() {
        return "\nDados da Pessoa Juridica: " + 
               super.toString() +
               "\nCNPJ: " + cnpj + 
               "\nInscricao Estadual: " + inscricaoEstadual;
    }
    
    
    
    
}

File: Python/04- POO/03- Atividades Pontuadas/Atividade 01/models/juridica.py
from models.pessoa import Pessoa
from models.endereco import Endereco

class Juridica(Pessoa):
    def __init__(self, nome, telefone, email, endereco: Endereco, cnpj: str, inscricao_estadual: str) ->None:
        super().__init__(nome, telefone, email, endereco)
        self.cnpj = cnpj
        self.inscricao_estadual = inscricao_estadual

    def __str__(self) -> str:
        return ( 
                f"\nCNPJ: {self.cnpj}"  
                f"\nInscrição Estadual: {self.inscricao_estadual}")